src/yaml/serialization.cr

```diff
@@ -138,7 +138,9 @@ module YAML
       end
     end
 
-    def initialize(ctx : YAML::ParseContext, node : ::YAML::Nodes::Node, dummy : Nil)
+    def initialize(*, context_for_yaml_serializable : YAML::ParseContext, node_for_yaml_serializable : ::YAML::Nodes::Node)
+      ctx = context_for_yaml_serializable
```

Comment:
You could avoid the local assignations by using external/internal naming of the parameters:

```crystal
def foo(*, external_long_name name : String, external_long_age age : Int32)
  {name, age}
end

foo(external_long_name: "John", external_long_age: 30) # => {"John", 30}
```

Reply:
What? never see this before